packages/next/shared/lib/dynamic.tsx

```diff
@@ -110,22 +113,33 @@ export default function dynamic<P = {}>(
   // Support for passing options, eg: dynamic(import('../hello-world'), {loading: () => <p>Loading something</p>})
   loadableOptions = { ...loadableOptions, ...options }
 
+  const suspenseOptions = loadableOptions as LoadableSuspenseOptions<P>
+  if (!process.env.__NEXT_CONCURRENT_MODE) {
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      suspenseOptions.suspense &&
+      !isServerSide
+    ) {
+      console.warn(
```

Comment:
Do we want this to be an error now?

Reply:
Add a error will be thrown when using suspense not rendering under `reactRoot`